aQute.libg/src/aQute/lib/io/IO.java

```diff
@@ -1157,6 +1171,40 @@ public static PrintWriter writer(OutputStream out, Charset encoding) {
 		return new PrintWriter(new OutputStreamWriter(out, encoding));
 	}
 
+	static class AppendableWriterAdapter extends Writer {
+
+		private Appendable appendable;
+
+		public AppendableWriterAdapter(Appendable appendable) {
+			this.appendable = appendable;
+		}
+
+		@Override
+		public void write(char[] cbuf, int off, int len) throws IOException {
+			appendable.append(String.valueOf(cbuf), off, len);
```

Comment:
Maybe `appendable.append(String.valueOf(cbuf, off, len));`? That is, pass `off` and `len` to `String.valueOf`.